fs/smartfs/smartfs_utils.c

```diff
@@ -671,7 +670,7 @@ int smartfs_finddirentry(struct smartfs_mountpt_s *fs,
 
                           memset(direntry->name, 0,
                                  fs->fs_llformat.namesize + 1);
-                          strncpy(direntry->name, entry->name,
+                          strlcpy(direntry->name, entry->name,
                                   fs->fs_llformat.namesize);
```

Comment:
```suggestion
                                  fs->fs_llformat.namesize + 1);
```
and remove line 671-672